security/advancedtls/advancedtls.go

```diff
@@ -193,6 +189,10 @@ type ClientOptions struct {
 	// By default, the maximum version supported by this package is used,
 	// which is currently TLS 1.3.
 	MaxVersion uint16
+	// serverNameOverride is for testing only. If set to a non-empty string,
+	// it will override the virtual host name of authority (e.g. :authority
```

Comment:
Also the SNI and the target hostname that we used during server cert verification. It might be worth commenting on these as well. :)

Reply:
To make sure I'm clear, you're saying to comment something like "it will override the virtual host name of authority... and the target hostname used during server cert verification"

Reply:
Correct.

Reply:
Added the comment